engine/src/main/java/org/terasology/config/RenderingConfig.java

```diff
@@ -423,65 +543,79 @@ public int getFrameLimit() {
     }
 
     public void setFrameLimit(int frameLimit) {
+        int oldLimit = this.frameLimit;
         this.frameLimit = frameLimit;
+        propertyChangeSupport.firePropertyChange(FRAME_LIMIT, oldLimit, this.frameLimit);
     }
 
     public int getFboScale() {
         return fboScale;
     }
 
     public void setFboScale(int fboScale) {
+        int oldScale = this.fboScale;
         this.fboScale = fboScale;
+        propertyChangeSupport.firePropertyChange(FBO_SCALE, oldScale, this.fboScale);
     }
 
     public boolean isClampLighting() {
         return clampLighting;
     }
 
     public void setClampLighting(boolean clampLighting) {
+        boolean oldValue = this.clampLighting;
         this.clampLighting = clampLighting;
+        propertyChangeSupport.firePropertyChange(CLAMP_LIGHTING, oldValue, this.clampLighting);
     }
 
     public ScreenshotSize getScreenshotSize() {
         return screenshotSize;
     }
 
     public void setScreenshotSize(ScreenshotSize screenshotSize) {
+        ScreenshotSize oldSize = this.screenshotSize;
         this.screenshotSize = screenshotSize;
+        propertyChangeSupport.firePropertyChange(SCREENSHOT_SIZE, oldSize, this.screenshotSize);
+        // propertyChangeSupport fires only if oldObject != newObject.
+        // This method could theoretically use a better equality check then. In practice
+        // it's unlikely a new ScreenshotSize instance will ever be value-per-value identical
+        // to the previous one, not requiring the event to be fired.
     }
 
     public String getScreenshotFormat() {
         return screenshotFormat;
     }
 
     public void setScreenshotFormat(String screenshotFormat) {
-        this.screenshotFormat = screenshotFormat;
+
+        // propertyChangeSupport fires if oldFormat != newFormat which in the case of strings is always true.
```

Comment:
@msteiger: Dear Martin, as Immortius is no longer as involved in the project as he used to, I'd be happy to refactor this again, in accordance with your preferences. However, I need an example pattern of what you'd like for a property. 

Also, @tdgunes will hopefully start using this functionality soon, to dynamically add and remove nodes from the RenderGraph as the config properties are modified. So, I'd like to deal with this quickly.